lib/knapsack_pro/extensions/rspec_extension.rb

```diff
@@ -90,15 +90,20 @@ def knapsack__run_specs(queue_runner)
 
           configuration.reporter.report(_expected_example_count = 0) do |reporter|
             configuration.with_suite_hooks do
-              queue_runner.with_batch do |test_file_paths|
+              queue_runner.with_batch do |test_file_paths, store|
                 knapsack__load_spec_files_batch(test_file_paths)
 
                 examples_passed = ordering_strategy.order(world.example_groups).map do |example_group|
                   queue_runner.handle_signal!
                   example_group.run(reporter)
                 end.all?
 
-                node_examples_passed = false unless examples_passed
+                if examples_passed
+                  store.last_batch_passed!
+                else
+                  store.last_batch_failed!
+                  node_examples_passed = false
+                end
```

Comment:
FYI `node_examples_passed = true` is default set before any batch is executed. 
The `node_examples_passed = false` must be set only when tests from a batch failed (only when at least one batch failed). 

If we use `node_examples_passed = examples_passed` and the last batch passed and the previous one not then it would lead to CI node being marked as passed which is not true.

Reply:
I hesitated to use something like `store.set_last_batch_status(examples_passed)` because then you must know that we accept true/false as argument. 

In case of some test runners like Cucumber the result from the test runner is exit code so you need to convert it to boolean before you pass it to `store.set_last_batch_status(examples_passed)`.

I decided to use `store.last_batch_passed!` or `store.last_batch_failed!` because it felt more explicit.

What do you think?

Reply:
You are right. Let's disregard my comment here.